app/models/dataset.rb

```diff
@@ -105,7 +105,7 @@ def uuid_and_title
   end
 
   def should_generate_new_friendly_id?
-    title_changed?
+    name.blank?
```

Comment:
We generate a new slug only if the `name` field (where we store slugs) is blank. That way, the slugs from legacy stay the same in Beta, and we avoid overwriting the slugs in Legacy when we sync datasets from Beta => Legacy.

The cost of this (for now): changing the title of a dataset in Beta does not generate a new slug.